openedx/core/djangolib/fields.py

```diff
@@ -26,3 +28,27 @@ def get_db_prep_value(self, value, connection, prepared=False):
             return None
         else:
             return value
+
+
+class BigAutoField(models.AutoField):
+    """
+    AutoField that uses BigIntegers.
+
+    This exists in Django as of version 1.10.
+    """
+
+    def db_type(self, connection):
+        """
+        The type of the field to insert into the database.
+
+        This may need to be modified if used with non-MySQL databases
+        """
+        return "bigint AUTO_INCREMENT"
+
+    def rel_db_type(self, connection):
+        """
+        The type to be used by relations pointing to this field.
+
+        Not used until Django 1.10.
+        """
+        return "bigint"
```

Comment:
Hi @jcdyer ! Questions:
- Will this work with sqlite3, which is typically used in unittests?
- Is the Django version unsigned or signed? Unsigned makes more sense here, of course.
- Post-Django-upgrade, the initial migration will still point to this file's def. Do you advise just adding an import of BigAutoField to the top of this file post-upgrade?
- What's the back-of-envelope calculation for when edx.org will get > 4.2 billion of these completions?

Reply:
@doctoryes Good point. `BIGINT` is allowed for sqlite, but the current combination is causing the python test suite to fail with this sqlite error:

```
OperationalError: AUTOINCREMENT is only allowed on an INTEGER PRIMARY KEY
```

The [django BigAutoField](https://docs.djangoproject.com/en/1.10/_modules/django/db/models/fields/#BigAutoField) is signed. It uses [`bigint AUTO_INCREMENT` on MySQL](https://github.com/django/django/blob/dd82f3327124fd2762cf6df2ac8c6380772bf127/django/db/backends/mysql/base.py#L105) and [`integer AUTOINCREMENT` on sqlite](https://github.com/django/django/blob/d0c761d3f84aa340175b5f8646d7b2f12d2e75be/django/db/backends/sqlite3/base.py#L55-L82) which works around the above issue, since sqlite uses 64-bit primary keys in any case and allows up to 8 byte ints in any column anyways regardless of its declared type.

@jcdyer I'll give a final review once the tests are fixed.

Reply:
Looks like @doctoryes has asked most of the questions I would care about.

Reply:
@doctoryes The back-of-napkin calculation that @bradenmacdonald came up with was 10M users taking 3 courses eaches with 100 completable blocks per course = 3B entries.  It will probably take longer than that to get there for a couple reasons:  

1. We won't (and can't) backfill this information.  
2. We aren't currently planning to fill empty records for all blocks when a user enrolls in a course so never-visited blocks are non-records (but this could change).  

So we aren't likely to actually hit the cap for significantly longer if at all, but it's close enough that I think a better-safe-than-sorry policy is warranted.